nautobot_netbox_importer/diffsync/adapters/netbox.py

```diff
@@ -17,11 +19,126 @@ class NetBox210DiffSync(N2NDiffSync):
 
     logger = structlog.get_logger()
 
+    _unsupported_fields = {}
+
     def __init__(self, *args, source_data=None, **kwargs):
         """Store the provided source_data for use when load() is called later."""
         self.source_data = source_data
         super().__init__(*args, **kwargs)
 
+    @property
+    def unsupported_fields(self):
+        """Public interface for accessing class attr `_unsupported_fields`."""
+        return self.__class__._unsupported_fields  # pylint: disable=protected-access
+
+    @staticmethod
+    def _get_ignored_fields(netbox_data: Dict, nautobot_instance: NautobotBaseModel) -> Set[str]:
+        """
+        Get fields from NetBox JSON that were not handled by the importer.
+
+        This only counts fields that have values.
+
+        Args:
+            netbox_data: The NetBox data for a particular database entry.
+            nautobot_instance: The Nautobot DiffSync instance created for `netbox_data`.
+
+        Returns:
+            set: The NetBox field names ignored by the importer.
+        """
+        # Get fields passed from NetBox that have values and ignore internal fields
+        netbox_fields = {key for key, value in netbox_data.items() if value and not key.startswith("_")}
+        # Get fields set on the model instance
+        instance_fields = nautobot_instance.__fields_set__
+        # Account for aliases when gettig a diff of fields instantiated on the model
+        field_aliases = {field.alias for field in nautobot_instance.__fields__.values() if field.alias != field.name}
+        return netbox_fields - instance_fields - field_aliases - nautobot_instance.ignored_fields
+
+    def _log_ignored_fields_details(
+        self,
+        netbox_data: Dict,
+        nautobot_instance: NautobotBaseModel,
+        model_name: str,
+        ignored_fields: Set[str],
+    ) -> None:
+        """
+        Log a debug message for NetBox fields ignored by the importer.
+
+        This will log for every instance of fields that were ignored by the
+        importer, so if there are a 100 instances of a model with an ignored
+        field, then 100 log entries will be generated. In order to prevent the
+        logs from generating too much noise, this is only logged as a debug.
+
+        Args:
+            netbox_data: The NetBox data for a particular database entry.
+            nautobot_instance: The Nautobot DiffSync instance created for `netbox_data`.
+            model_name: The DiffSync modelname for the NetBox entry.
+            ignored_fields: The field names in `netbox_data` that were ignored.
+        """
+        ignored_fields_with_values = (f"{field}={netbox_data[field]}" for field in ignored_fields)
+        ignored_fields_data_str = ", ".join(ignored_fields_with_values)
+        self.logger.debug(
+            "NetBox field not defined for DiffSync Model",
+            comment=(
+                f"The following fields were defined in NetBox for {model_name}, "
```

Comment:
Wonder if this should include something like the model instance `identifier` as well so that the user can tell *which* instance of `model_name` was affected?

Reply:
I have the pk listed below, what would be the identifier?